crates/viewer/src/camera.rs

```diff
@@ -30,29 +51,60 @@ impl Camera {
         self.projection = Projection::Orthographic;
     }
 
+    fn get_local_frame(&self) -> Mat3 {
+        Mat3::from_quat(self.orientation)
+    }
+
+    /// Pan the camera view horizontally and vertically. Look-at target will move along with the
+    /// camera.
+    pub fn pan(&mut self, delta: Vec2) {
+        self.target -= self.get_local_frame() * delta.extend(0.0);
+    }
+
+    /// Zoom in or out, while looking at the same target.
+    pub fn zoom(&mut self, zoom_delta: f32) {
+        self.zoom_factor = f32::max(self.zoom_factor * f32::exp(zoom_delta), MIN_ZOOM_FACTOR);
+    }
+
+    /// Orbit around the target while keeping the distance.
+    pub fn rotate(&mut self, rotator: Quat) {
+        self.orientation = (self.orientation * rotator).normalize();
+    }
+
     pub fn matrix(&self) -> Mat4 {
         // These magic numbers are configured so that the particular model we are loading is
         // visible in its entirety. They will be dynamically computed eventually when we have "fit
         // to view" function or alike.
-        let proj = match self.projection {
-            Projection::Orthographic => Mat4::orthographic_rh(
-                -100.0 * self.aspect_ratio,
-                100.0 * self.aspect_ratio,
-                -100.0,
-                100.0,
-                -1000.0,
-                1000.0,
-            ),
+
+        let (proj, effective_radius) = match self.projection {
+            Projection::Orthographic => {
+                let proj = Mat4::orthographic_rh(
+                    -50.0 * self.zoom_factor * self.aspect_ratio,
+                    50.0 * self.zoom_factor * self.aspect_ratio,
+                    -50.0 * self.zoom_factor,
+                    50.0 * self.zoom_factor,
+                    -1000.0,
+                    1000.0,
+                );
+                (proj, self.radius)
+            },
             Projection::Perspective => {
-                Mat4::perspective_rh(std::f32::consts::PI / 2.0, self.aspect_ratio, 0.01, 1000.0)
+                let proj = Mat4::perspective_rh(
+                    std::f32::consts::PI / 2.0,
+                    self.aspect_ratio,
+                    10.0,
```

Comment:
I played around a bit with this, I think `1.0` is a bit more comfortable for most of the models I tried.